File: web/plugins/divide.js
import { WebPlugin } from "../engine/plugin.js";

export class DividePlugin extends WebPlugin {
    constructor() {
        super("割る");
    }

    async execute(line, context) {
        const match = line.match(/^割る\s+(.+?)\s+(.+)$/);

        if (!match) {
            throw new Error(`割るの書式が正しくありません: ${line}`);
        }

        const [, target, expression] = match;

        const value = context.parser.parseNumber(
            expression,
            context.runtime
        );

        if (value === 0) {
            throw new Error("0で割ることはできません。");
        }

        if (context.runtime.exists(target)) {
            const current = context.parser.parseNumber(
                target,
                context.runtime
            );

            context.runtime.set(target, current / value);
            return;
        }

        const current = context.parser.parseNumber(
            target,
            context.runtime
        );

        context.runtime.write(current / value);
    }
}

File: pc/plugins/divide.py
import re
from core.plugin import Plugin


class DividePlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'割る\s+(.+?)\s+(.+)', line)

        if not match:
            return None

        name = match.group(1)
        value = match.group(2)

        return f"{name} /= {value}"